pkg/pillar/types/assignableadapters.go

```diff
@@ -460,6 +506,19 @@ func (aa *AssignableAdapters) CheckBadAssignmentGroups(log *base.LogObject, PCIS
 				}
 				continue
 			}
+			if ib.UsbProduct != "" || ib2.UsbProduct != "" {
+				usbProduct1 := strings.TrimLeft(ib.UsbProduct, "0")
+				usbProduct2 := strings.TrimLeft(ib2.UsbProduct, "0")
+				if usbProduct1 == usbProduct2 {
```

Comment:
@christoph-zededa still necessary?

Reply:
The comment above the function doesn't say what it is expected to do with usbaddr comparison let alone usbproduct.

I think this is just double-checking that the controller didn't do bad assignments so that conflicting assignments are made when it comes to the assigngrp. (That is what the comment says about pciaddr).

But since for USB devices we compare all of pcilong, usbaddr, and usbproduct, only if all of them match do we have an assignment. Thus I think the logic needs to be diffent.

if usbAddrs are set {
   if usbAddrs are different {
       continue
   }
   save/append to conflict error string
}
if usbProducts are set {
  if usbProducts are different {
      continue
   }
   save/append to conflict error string
}
if ib.Pcilong are set in both {
   if !PCISameController {
      continue
   }
   save/append to conflict error string
}
// Save error error; set changed=true

That way only if all of the set fields match do we generate an error; otherwise we assume we can assign them separately.

Does that make sense or did I miss something?